lib/UserDefinedAbstract.php

```diff
@@ -0,0 +1,601 @@
+	 * any descendant nodes of the article item node. Attributes can be
+	 * selected by prepending an @ char before the attributes name.
+	 *
+	 * Use {@see UserDefinedAbstract::getExpressionItemTimestamp()} to read this parameter
+	 */
+	const USER_EXPRESSION_ITEM_TIMESTAMP = '';
+
+	/**
+	 * User expression for extracting item enclosures (media content like
+	 * images or movies) from the item context
+	 * This expression should match a node's attribute containing an article
+	 * image URL (usually the src attribute of an <img> tag or a style
+	 * attribute). It should start with a dot followed by two forward slashes,
+	 * referring to any descendant nodes of the article item node. Attributes
+	 * can be selected by prepending an @ char before the attributes name.
+	 *
+	 * Use {@see UserDefinedAbstract::getExpressionItemEnclosures()} to read this parameter
+	 */
+	const USER_EXPRESSION_ITEM_ENCLOSURES = '';
+
+	/**
+	 * User expression for extracting an item category from the item context
+	 * This expression should match a node or node's attribute contained
+	 * within each article item node containing the article category. This
+	 * could be inside <div> or <span> tags or sometimes be hidden
+	 * in a data attribute. It should start with a dot followed by two
+	 * forward slashes, referring to any descendant nodes of the article
+	 * item node. Attributes can be selected by prepending an @ char
+	 * before the attributes name.
+	 *
+	 * Use {@see UserDefinedAbstract::getExpressionItemCategories()} to read this parameter
+	 */
+	const USER_EXPRESSION_ITEM_CATEGORIES = '';
+
+	/**
+	 * User expression for extracting an item uid from the item content
+	 *
+	 * Use {@see UserDefinedAbstract::getExpressionItemUid()} to read this parameter
+	 */
+	const USER_EXPRESSION_ITEM_UID = '';
+
+	/**
+	 * Fix encoding
+	 * Set this to true for fixing feed encoding by invoking PHP's utf8_decode
+	 * function on all extracted texts. Try this in case you see "broken" or
+	 * "weird" characters in your feed where you'd normally expect umlauts
+	 * or any other non-ascii characters.
+	 *
+	 * Use {@see UserDefinedAbstract::getSettingFixEncoding()} to read this parameter
+	 */
+	const SETTING_FIX_ENCODING = false;
+
+	/**
+	 * Internal storage for resulting feed name, automatically detected
+	 * @var string
+	 */
+	private $feedName;
+
+	/**
+	 * Internal storage for resulting feed name, automatically detected
+	 * @var string
+	 */
+	private $feedUri;
+
+	/**
+	 * Internal storage for resulting feed favicon, automatically detected
+	 * @var string
+	 */
+	private $feedIcon;
+
+	public function getName() {
+		return $this->feedName ?: parent::getName();
+	}
+
+	public function getURI() {
+		return $this->feedUri ?: parent::getURI();
+	}
+
+	public function getIcon() {
+		return $this->feedIcon ?: parent::getIcon();
+	}
+
+	/**
+	 * Source URL
+	 * @return string
+	 */
+	protected function getSourceUrl(){
```

Comment:
I don't really see the purpose of having a class method that returns a static const.

Reply:
So that it can be overridden. For example, see [Google Groups Bridge](https://github.com/RSS-Bridge/rss-bridge/blob/master/bridges/GoogleGroupsBridge.php#L43=).